mobile/appcontainer/RefreshSwiper.js

```diff
@@ -0,0 +1,129 @@
+import {Icon} from '@xh/hoist/icon';
+import classNames from 'classnames';
+import {hoistCmp, HoistModel, uses, useLocalModel, XH} from '@xh/hoist/core';
+import {frame, div} from '@xh/hoist/cmp/layout';
+import {isFinite} from 'lodash';
+import {computed, observable, action, makeObservable} from '@xh/hoist/mobx';
+import {gestureDetector} from '@xh/hoist/kit/onsen';
+import {AppContainerModel} from '@xh/hoist/appcontainer/AppContainerModel';
+import './RefreshSwiper.scss';
+
+
+/**
+ * Wrap the Application content with drag gesture handling and a refresh icon.
+ *
+ * @private
+ */
+export const refreshSwiper = hoistCmp.factory({
+    model: uses(AppContainerModel),
+    render({model, children}) {
+        const impl = useLocalModel(() => new LocalModel(model));
+        return frame(
+            swipeIcon({model: impl}),
+            gestureDetector({
+                style: {display: 'flex', width: '100%', height: '100%'},
+                onDragStart: impl.onDragStart,
+                onDrag: impl.onDrag,
+                onDragEnd: impl.onDragEnd,
+                item: children
+            })
+        );
+    }
+});
+
+const swipeIcon = hoistCmp.factory(
+    ({model}) => {
+        if (!model.swipeStarted) return null;
+
+        const {swipeProgress, swipeComplete} = model,
+            top = -40 + (swipeProgress * 150),
```

Comment:
I feel like this comes down too far, and something like 100 would be better (i.e. just below the top navigation).